src/Behavioral.Automation/Bindings/PageScopeObserverBinding.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Behavioral.Automation.Services;
+using Behavioral.Automation.Services.Mapping.Contract;
+using JetBrains.Annotations;
+using TechTalk.SpecFlow;
+
+namespace Behavioral.Automation.Bindings
+{
+    /// <summary>
+    /// Allows to automatically switch PageContext, depending on current Driver url
+    /// </summary>
+    [Binding]
+    public class PageScopeObserverBinding
+    {
+        private readonly IScopeContextManager _scopeContextManager;
+        private readonly IDriverService _driverService;
+
+        public PageScopeObserverBinding([NotNull] IScopeContextManager scopeContextManager, [NotNull] IDriverService driverService)
+        {
+            _scopeContextManager = scopeContextManager;
+            _driverService = driverService;
+        }
+
+        [BeforeStep]
```

Comment:
Let's use a common place for all hooks?

Reply:
Like what? CommonHooks.cs? Won't it be hard debug in future, since this class will contain all sorts of bindings, that are not grouped by logic?

Reply:
We can create a folder `Hooks` and store there all hooks. It can be one or several classes. 
Hooks list ref: https://docs.specflow.org/projects/specflow/en/latest/Bindings/Hooks.html

Reply:
Done